### DISCUSSION TITLE
Are Rf neurons able to reproduce STFT

### DISCUSSION THREAD

**Discussion by zeinebCh** [lava-nc/lava-dl#194] [Q&A]

**Comment by bamsumit** [comment]
Yes, a single RF layer can extract something similar to STFT. [This paper](https://arxiv.org/pdf/2111.03746.pdf) might be useful. Your data format looks correct. Not sure why you would want to train it, it follows directly from the mathematical formulation.

**Comment by Michaeljurado42** [comment]
Hello all. I was debating whether to post this here or on this other [discussion](https://github.com/lava-nc/lava-dl/discussions/198).
However, since you mentioned the paper on this thread I thought I would ask it here. 

I am having some difficulty replicating the decoding process mentioned in the paper.

It says that the spiking STFT "can be inverted by convolving the spikes with the neuron's oscillation kernel and integrating across the population".  I attempted this and ran into a whole bunch of problems as I was trying to reproduce figure 4. 

The jupyter notebook I am using is [here](https://github.com/lava-nc/lava-dl/files/11696939/STFT_APPROXIMATION_R.F.zip). You can also view the [pdf](https://github.com/lava-nc/lava-dl/files/11696981/STFT_APPROXIMATION_R.F.pdf) if you would like.  I actually think that if we can get this to work it would be a great tutorial to add to lava-dl. (After cleaning it up a lot that is) 

Any guidance on this would be greatly appreciated.

**Reply by zeinebCh** [reply]
@Michaeljurado42 Hello, I really appreciate your willingness to share your code.
The resonating frequencies of the neuron are now logarithmic spaced. Do you know any idea of how to make them linearly spaced?
Thanks

**Reply by Michaeljurado42** [reply]
What you do is modify the code [here](https://github.com/lava-nc/lava-dl/blob/main/src/lava/lib/dl/slayer/neuron/rf.py#L366) to use linspace instead of logspace. This might be a good addition to add to the lava-dl codebase.
